src/librustdoc/clean/inline.rs

```diff
@@ -165,9 +166,8 @@ crate fn record_extern_fqn(cx: &mut DocContext<'_>, did: DefId, kind: ItemType)
     let crate_name = cx.tcx.crate_name(did.krate).to_string();
 
     let relative = cx.tcx.def_path(did).data.into_iter().filter_map(|elem| {
-        // extern blocks have an empty name
-        let s = elem.data.to_string();
-        if !s.is_empty() { Some(s) } else { None }
+        // Filter out extern blocks
+        (elem.data != DefPathData::ForeignMod).then(|| elem.data.to_string())
```

Comment:
cc #91948: How does this interact with that PR, specifically https://github.com/rust-lang/rust/pull/91948#discussion_r771647111?

Reply:
When both PRs are merged this will turn into just `elem.data.get_opt_name()`.